cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs

```diff
@@ -211,3 +222,84 @@ pub unsafe extern "C" fn ecall_key_gen(
     trace!("ecall_key_gen key pk: {:?}", public_key.to_vec());
     sgx_status_t::SGX_SUCCESS
 }
+
+#[cfg(not(feature = "SGX_MODE_HW"))]
+fn verify_local_report(_cert: &[u8]) -> sgx_status_t {
+    sgx_status_t::SGX_SUCCESS
+}
+
+#[cfg(feature = "SGX_MODE_HW")]
+fn verify_local_report(cert: &[u8]) -> sgx_status_t {
+    let report = match AttestationReport::from_cert(cert) {
+        Ok(r) => r,
+        Err(_) => {
+            error!("Error parsing report");
+            return sgx_status_t::SGX_ERROR_INVALID_PARAMETER;
+        }
+    };
+
+    let node_auth_result = NodeAuthResult::from(&report.sgx_quote_status);
+    // print
+    match verify_quote_status(&report.sgx_quote_status) {
+        Err(status) => match status {
+            NodeAuthResult::SwHardeningAndConfigurationNeeded => {
+                println!("Platform status is SW_HARDENING_AND_CONFIGURATION_NEEDED. This means is updated but requires further BIOS configuration");
+            }
+            NodeAuthResult::GroupOutOfDate => {
+                println!("Platform status is GROUP_OUT_OF_DATE. This means that one of the system components is missing a security update");
+            }
+            _ => {
+                println!("Platform status is {}", status);
+            }
+        },
+        _ => println!("Platform Okay!"),
+    }
+
+    // print platform blob info
+    match node_auth_result {
+        NodeAuthResult::GroupOutOfDate | NodeAuthResult::SwHardeningAndConfigurationNeeded => unsafe {
+            if let Some(platform_info) = report.platform_info_blob {
+                let mut update_info = sgx_update_info_bit_t::default();
+                let mut rt = sgx_status_t::default();
+                let res = ocall_get_update_info(
+                    &mut rt as *mut sgx_status_t,
+                    platform_info[4..].as_ptr() as *const sgx_platform_info_t,
+                    1,
+                    &mut update_info as *mut sgx_update_info_bit_t,
+                );
+
+                if res != sgx_status_t::SGX_SUCCESS {
+                    println!("res={:?}", res);
+                    return res;
+                }
+
+                if rt != sgx_status_t::SGX_SUCCESS {
+
+                    if update_info.ucodeUpdate != 0 {
+                        println!("Processor Firmware Update (ucodeUpdate). A security upgrade for your computing
+                            device is required for this application to continue to provide you with a high degree of
+                            security. Please contact your device manufacturer’s support website for a BIOS update
+                            for this system");
```

Comment:
Is the indentation omitted by the compiler?
I think you need to add `\` or something.

Reply:
> Just a few small notes, I know that this is still not finished :)
> Also, there are a bunch of small formatting adjustments. Were those applied automatically by rustfmt?

yep

Reply:
Idk it looked fine when it was printed out

Reply:
ah, cool